### PR TITLE
Add aiofile

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by BastianZim** [comment]
@conda-forge/help-python Would be ready for review, thanks!

File: recipes/aiofile/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
+  run:
+    - caio ~=0.9.0
```
Comment (jaimergp) [inline recipes/aiofile/meta.yaml:23]: I think this way of specifying constrained is not fully supported now (and in fact we patch it for backwards compatibility). Can you elaborate on why this is needed?

Comment (BastianZim) [reply recipes/aiofile/meta.yaml]: It's from upstream: https://github.com/mosquito/aiofile/blob/9915207c8564c4b49d57ce16003c385a42aa935b/setup.py#L61

But yeah you're right, completely forgot that it's not supported. Have translated it to conda specs.

Comment (BastianZim) [reply recipes/aiofile/meta.yaml]: ICYI https://github.com/conda-incubator/grayskull/issues/270

**Comment by BastianZim** [comment]
Thanks!

**Pull request merged**